task_processing/plugins/mesos/mesos_executor.py

```diff
@@ -53,10 +53,11 @@ class MesosExecutor(TaskExecutor):
     def __init__(
         self,
         role,
-        authentication_principal='taskproc',
-        credential_secret_file=None,
+        principal='taskproc',
+        secret=None,
         mesos_address='127.0.0.1:5050',
         translator=mesos_status_to_event,
+        framework_kwargs={},
```

Comment:
in general I'm in favour of being explicit with parameters, rather than asking the caller to go looking through the code to find out what parameters they can set. I'd make each of the keys you set the default for explicit in here.